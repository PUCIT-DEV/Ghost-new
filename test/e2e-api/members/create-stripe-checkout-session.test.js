const {agentProvider, mockManager, fixtureManager, matchers} = require('../../utils/e2e-framework');

let membersAgent, adminAgent, membersService;

describe('Create Stripe Checkout Session', function () {
    before(async function () {
        const agents = await agentProvider.getAgentsForMembers();
        membersAgent = agents.membersAgent;
        adminAgent = agents.adminAgent;

        membersService = require('../../../core/server/services/members');

        await fixtureManager.init('members');
        await adminAgent.loginAsOwner();
    });

    beforeEach(function () {
        mockManager.mockMail();
        mockManager.mockStripe();
    });

    afterEach(function () {
        mockManager.restore();
    });

    it('Does not allow to create a checkout session if the custoemrEmail is associated with a paid member', async function () {
        const {body: {tiers}} = await adminAgent.get('/tiers/?include=monthly_price&yearly_price');

        const paidTier = tiers.find(tier => tier.type === 'paid');

        await membersAgent.post('/api/create-stripe-checkout-session/')
            .body({
                customerEmail: 'paid@test.com',
                tierId: paidTier.id,
                cadence: 'month'
            })
            .expectStatus(403)
            .matchBodySnapshot({
                errors: [{
                    id: matchers.anyUuid,
                    code: 'CANNOT_CHECKOUT_WITH_EXISTING_SUBSCRIPTION'
                }]
            });
    });
});
