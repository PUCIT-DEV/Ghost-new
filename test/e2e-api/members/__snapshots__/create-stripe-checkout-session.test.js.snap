// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Stripe Checkout Session Does allow to create a checkout session if the customerEmail is not associated with a paid member 1: [body] 1`] = `
Object {
  "publicKey": "pk_test_for_stripe",
  "sessionId": "cs_123",
}
`;

exports[`Create Stripe Checkout Session Does allow to create a checkout session if the customerEmail is not associated with a paid member 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "*",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-type": "application/json",
  "vary": "Accept-Encoding",
  "x-powered-by": "Express",
}
`;

exports[`Create Stripe Checkout Session Does not allow to create a checkout session if the customerEmail is associated with a paid member 1: [body] 1`] = `
Object {
  "errors": Array [
    Object {
      "code": "CANNOT_CHECKOUT_WITH_EXISTING_SUBSCRIPTION",
      "context": null,
      "details": null,
      "ghostErrorCode": null,
      "help": null,
      "id": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
      "message": "A subscription exists for this Member.",
      "property": null,
      "type": "NoPermissionError",
    },
  ],
}
`;

exports[`Create Stripe Checkout Session Does not allow to create a checkout session if the customerEmail is associated with a paid member 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "*",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": "268",
  "content-type": "application/json; charset=utf-8",
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Encoding",
  "x-powered-by": "Express",
}
`;
